pkg/authorization/authorizer/scope/converter.go

```diff
@@ -384,45 +419,46 @@ func (e clusterRoleEvaluator) ResolveGettableNamespaces(scope string, clusterPol
 		ResourceRequest: true,
 	}
 
-	errors := []error{}
-	for _, rule := range rules {
-		matches, err := authorizer.RuleMatches(attributes, rule)
-		if err != nil {
-			errors = append(errors, err)
-			continue
-		}
-		if matches {
-			return []string{scopeNamespace}, nil
-		}
+	if authorizerrbac.RulesAllow(attributes, rules...) {
+		return []string{scopeNamespace}, nil
 	}
 
-	return []string{}, kutilerrors.NewAggregate(errors)
+	return []string{}, nil
 }
 
 // TODO: direct deep copy needing a cloner is something that should be fixed upstream
 var localCloner = conversion.NewCloner()
 
+func remove(set []string, foreign string) []string {
+	newset := make([]string, 0, len(set))
+	for _, element := range set {
+		if element != foreign {
+			newset = append(newset, element)
+		}
+	}
+	return newset
+}
+
 // removeEscalatingResources inspects a PolicyRule and removes any references to escalating resources.
 // It has coarse logic for now.  It is possible to rewrite one rule into many for the finest grain control
 // but removing the entire matching resource regardless of verb or secondary group is cheaper, easier, and errs on the side removing
 // too much, not too little
-func removeEscalatingResources(in authorizationapi.PolicyRule) authorizationapi.PolicyRule {
-	var ruleCopy *authorizationapi.PolicyRule
+func removeEscalatingResources(in rbac.PolicyRule) rbac.PolicyRule {
+	var ruleCopy *rbac.PolicyRule
 
-	apiGroups := getAPIGroupSet(in)
 	for _, resource := range escalatingScopeResources {
-		if !(apiGroups.Has(resource.Group) && in.Resources.Has(resource.Resource)) {
+		if !(has(getAPIGroupLegacy(in), resource.Group) && has(in.Resources, resource.Resource)) {
 			continue
 		}
 
 		if ruleCopy == nil {
 			// we're using a cache of cache of an object that uses pointers to data.  I'm pretty sure we need to do a copy to avoid
 			// muddying the cache
-			ruleCopy = &authorizationapi.PolicyRule{}
-			authorizationapi.DeepCopy_authorization_PolicyRule(&in, ruleCopy, localCloner)
+			ruleCopy = &rbac.PolicyRule{}
+			rbac.DeepCopy_rbac_PolicyRule(&in, ruleCopy, localCloner)
 		}
 
-		ruleCopy.Resources.Delete(resource.Resource)
+		ruleCopy.Resources = remove(ruleCopy.Resources, resource.Resource)
```

Comment:
this is reallocating the set on each pass, even if the item is not found. we're already working on a copy, so we can mutate the original. prefer a removal that does not cause an allocation